Makefile

```diff
@@ -0,0 +1,34 @@
+CHANNEL ?= unpublished
+CHARM := sriov-cni
+
+ifndef NAMESPACE
+$(error NAMESPACE is not set)
```

Comment:
I would expect this to be required for `make upload`, but not for `make charm`. Can we make this check apply only to the targets that use it?